File: app/src/main/java/com/github/haocen2004/login_simulation/data/database/sponsor/SponsorDao.java
package com.github.haocen2004.login_simulation.data.database.sponsor;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface SponsorDao {
    @Insert
    void insertSponsors(SponsorData... sponsorDatas);

    @Update
    void updateSponsors(SponsorData... sponsorData);

    @Delete
    void deleteSponsors(SponsorData... sponsorData);

    @Query("DELETE FROM SPONSORS")
    void deleteAllSponsors();

    @Query("SELECT * FROM SPONSORS ORDER BY ID DESC")
    List<SponsorData> getAllSponsors();
}

File: app/src/main/java/com/github/haocen2004/login_simulation/data/sponsor/database/SponsorDao.kt
package com.github.haocen2004.login_simulation.data.sponsor.database

import androidx.room.*

@Dao
interface SponsorDao {
    @Insert
    fun insertSponsors(vararg sponsorDatas: SponsorData?)

    @Update
    fun updateSponsors(vararg sponsorData: SponsorData?)

    @Delete
    fun deleteSponsors(vararg sponsorData: SponsorData?)

    @Query("DELETE FROM SPONSORS")
    fun deleteAllSponsors()

    @get:Query("SELECT * FROM SPONSORS ORDER BY ID DESC")
    val allSponsors: List<SponsorData>

    @Query("SELECT * FROM SPONSORS ORDER BY ID DESC LIMIT :limit OFFSET :offset")
    suspend fun getPagedList(limit: Int?, offset: Int?): List<SponsorData>
}